woocommerce/woocommerce-ios
# Lightweight storefront: layout switcher

Closes #11348 

## Description


<img width="888" alt="Screenshot 2023-12-14 at 17 49 46" src="https://github.com/woocommerce/woocommerce-ios/assets/266376/3739155c-3220-4c83-ba45-dc8b517ee380">

This PR adds the View for previewing theme.

Additionally, it also adds functionality to switch the device layout from mobile to tablet to desktop. This will automatically update the layout of the webpage responsively to match the selected device layout.

Other buttons on the screen do not work yet.

The layout currently missed the page selector in the middle, since there's since some UI discussion around it (namely, the iOS toolbar doesn't easily fit them).

## Testing instructions


To test, go to Menu > Settings > Themes. Once the carousel is loaded, tap on a theme. The previewer should show up.

### Video:

(ignore the height of the view shown here, it will be shown as a sheet instead as per design and the height issue can be tackled later)

https://github.com/woocommerce/woocommerce-ios/assets/266376/c26fd942-227b-49bf-85fc-ccbce3937320
